### ISSUE TITLE
Ctrl-w does not work in popup window

### ISSUE THREAD

**Issue by simaoneves** [junegunn/fzf.vim#1176]
- [x] I have fzf 0.23.0 or above
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [ ] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

The problem is simple, using `:Files` command (at least, have not tried others) always allowed me to input `ctrl-w` to delete the last word i am searching for.
After the default of opening FZF is now `window`, i cannot do it anymore.

**Comment by junegunn** [comment]
It works just fine. You should check your `tnoremap`s.

**Comment by simaoneves** [comment]
I don't have any `tmap` defined and i can reproduce this in the minimal configuration given (`vim -Nu <(curl https://gist.githubusercontent.com/junegunn/6936bf79fedd3a079aeb1dd2f3c81ef5/raw)`)

I also get the error `E994: Not allowed in a popup window` when trying to `ctrl-w` in the popup window.

**Comment by junegunn** [comment]
I see, the problem is only reproducible in regular Vim and I haven't noticed it because I use Neovim.

When you open up a terminal in Vim with `:terminal`, you'll realize that CTRL-W is not working as expected in there as well. And the workaround is to set `termwinkey` to a different key, say `set termwinkey=<c-z>`.

Do you think it would be a good idea for fzf to automatically set this option?

**Comment by sim590** [comment]
I'M having the same issue. Even though I `set termwinkey=<c-z>`, I'm getting the same errors (not allowed in popup window). Other hotkeys that used to work also don't work anymore like some of the readline style keys like `alt+b`, `alt+d`, `, etc. However, `ctrl+u`, `ctrl+a` and `ctrl+e` do work for instance.

**Comment by junegunn** [comment]
I'm not sure why the global `termwinkey` setting is not applied to the terminal buffer created by fzf. Could be a bug of Vim, or maybe I'm not using `term_start` correctly.

Anyway, with this patch,

```diff
diff --git a/plugin/fzf.vim b/plugin/fzf.vim
index 5fa555a..e59196d 100644
--- a/plugin/fzf.vim
+++ b/plugin/fzf.vim
@@ -836,6 +836,7 @@ function! s:execute_term(dict, command, temps) abort
         let term_opts.curwin = 1
       endif
       let fzf.buf = term_start([&shell, &shellcmdflag, command], term_opts)
+      call setbufvar(fzf.buf, '&termwinkey', '<c-z>')
       if is_popup && exists('#TerminalWinOpen')
         doautocmd <nomodeline> TerminalWinOpen
       endif
```

CTRL-W will work as expected.

> some of the readline style keys like alt+b, alt+d, , etc.

They already work just fine even without the patch. So it could be a configuration issue. See if you can reproduce the problem with this minimal config file.

```sh
vim -Nu <(curl https://gist.githubusercontent.com/junegunn/6936bf79fedd3a079aeb1dd2f3c81ef5/raw)
```

**Comment by sim590** [comment]
> Anyway, with this patch,
>
> ```diff
> diff --git a/plugin/fzf.vim b/plugin/fzf.vim
> index 5fa555a..e59196d 100644
> --- a/plugin/fzf.vim
> +++ b/plugin/fzf.vim
> @@ -836,6 +836,7 @@ function! s:execute_term(dict, command, temps) abort
>          let term_opts.curwin = 1
>        endif
>        let fzf.buf = term_start([&shell, &shellcmdflag, command], term_opts)
> +      call setbufvar(fzf.buf, '&termwinkey', '<c-z>')
>        if is_popup && exists('#TerminalWinOpen')
>          doautocmd <nomodeline> TerminalWinOpen
>        endif
> ```
> CTRL-W will work as expected.

Ah yes. I can confirm. I was just trying to set this option in my `.vimrc` and didn't work, but the way you set it there, it works.

> They already work just fine even without the patch. So it could be a
> configuration issue. See if you can reproduce the problem with this minimal
> config file.

You are right. This was due to the vim-rsi plugin. I decided to abandon this plugin as it doesn't bring much anyway. Thank you for fast clarification and solution!

**Comment by simaoneves** [comment]
This looks great! Thanks for the help and fixing this issue 😄

**Comment by Caid11** [comment]
Just a heads up, older versions of vim don't have the "termwinkey" option, even though they have terminal support. For example, my version (8.0, the default with Ubuntu 18.04) throws this error message with the above patch:
```
Error detected while processing function fzf#vim#buffers[3]..<SNR>64_fzf[20]..fzf#run[59]..<SNR>30_execute_term:
line   75:
E355: Unknown option: termwinkey
```

Guarding it like so prevents the error for me:
```
      if exists('&termwinkey')
          call setbufvar(fzf.buf, '&termwinkey', '<c-z>')
      endif
```

**Comment by junegunn** [comment]
Thanks, fixed in 7acdaf0b4335a6c089bce5b73ff23aeb3b9d9c92.
